### DISCUSSION TITLE
How to get per channel luminance data?

### DISCUSSION THREAD

**Discussion by nixcamic** [SixLabors/ImageSharp#1309] [Q&A] [answered]
Is there an easy way to get per channel luminance (brightness) data in ImageSharp? In Magick.net I'd do something like: 

```c#
Statistics stats = image.Statistics();
double red = stats.GetChannel(PixelChannel.Red).Mean;
```

**Comment by nixcamic** [comment]
On a completely different subject, how do I get github discussions to not ignore the newline between those two lines of code in my example?

**Reply by JimBobSquarePants** [reply]
https://github.github.com/gfm/#fenced-code-blocks

**Reply by nixcamic** [reply]
Awesome thanks!

**Comment by JimBobSquarePants** [comment, marked as answer]
That example you've cited from Magick.NET isn't luminance by my reckoning, merely an average of the red component.

Luminance is the weighted sum of linear RGB components.

See https://github.com/SixLabors/ImageSharp/discussions/1287 for an example of determining the Luma (weighted sum of gamma compressed RGB components). This is the most common supposition of brightness.

**Reply by nixcamic** [reply]
Thanks, rephrasing it then, cause I'm not really sure on the terminology I guess. Is there a way to get that average of the red (or any other) component? I already have it working with those values in Magick.net.

Also apologies for reediting my posts over and over, still trying to figure out github discussions.

**Reply by JimBobSquarePants** [reply]
No worries, it's new tech...

It's actually pretty simple. You could calculate it like the following:

```c#
long red = 0;
double averageRed = 0;
using (var image = Image.Load<Rgba32>(...))
{
    for (int y = 0; y < image.Height; y++)
    {
        Span<Rgba32> rowSpan = image.GetPixelRowSpan(y);
        for (int x = 0; x < rowSpan.Length; x++)
        {
            red += rowSpan[x].R;
        }
    }

    // Finally calculate the average red.
    averageRed = red / (double)(image.Width * image.Height);
}
```

There reason there is no built-in method like Magick.NET is because our methods have to be pixel format agnostic since they can be any value determined by the type parameter given to `Image<TPixel>` .

**Reply by nixcamic** [reply]
Awesome, does exactly what I needed. With that it only took me like 20 minutes to port my app from Magick.net to ImageSharp. Granted it's not a huge app, but now that everything is managed it's a lot easier to make it cross platform.
